sinks/sinks_service.go

```diff
@@ -71,13 +72,29 @@ func (svc sinkService) encryptMetadata(sink Sink) (Sink, error) {
 	sink.Config.FilterMap(func(key string) bool {
 		return key == backend.ConfigFeatureTypePassword
 	}, func(key string, value interface{}) (string, interface{}) {
-		newValue, err2 := svc.passwordService.EncodePassword(value.(string))
+		var stringVal string
+		switch v := value.(type) {
+		case *string:
+			stringVal = *v
+		case string:
+			stringVal = v
+		}
+		newValue, err2 := svc.passwordService.EncodePassword(stringVal)
 		if err2 != nil {
 			err = err2
 			return key, value
 		}
 		return key, newValue
 	})
+	if sink.ConfigData != "" {
+		sinkBE := backend.GetBackend(sink.Backend)
+		sink.ConfigData, err = sinkBE.ConfigToFormat(sink.Format, sink.Config)
```

Comment:
probably should check that sinkBE is not nil

Reply:
actually, at this point, the backend would have failed earlier.
when it is called by the `CreateSink`, it has a validation in the sink_service.go:35~38 iirc.
if it is called from `UpdateSink` the backend is already estabilished.

Reply:
understood, but I find the code will be more reliable if you add a simple check even here. code has a tendency to be e.g. copy/pasted to other places for example, where that may not be true.

Reply:
commit with fixes
https://github.com/orb-community/orb/pull/2268/commits/a0b7711187ad405b112fd2de0bbc4c4b38d85767